service-registry/src/main/java/org/apache/servicecomb/serviceregistry/client/IpPortManager.java

```diff
@@ -61,6 +67,7 @@ public IpPortManager(ServiceRegistryConfig serviceRegistryConfig, InstanceCacheM
     }
     int initialIndex = new Random().nextInt(defaultIpPort.size());
     currentAvailableIndex = new AtomicInteger(initialIndex);
+    maxRetryTimes = defaultIpPort.size() + (getDiscoveredIpPort() == null ? 0 : getDiscoveredIpPort().size());
```

Comment:
This code is not correct. maxRetryTimes is dynamically change and you can not getDiscoveryIpPort when initialize IpPortManager